### PR TITLE
Bug 1337822 make search engines.get unordered bundled engines more robust 

### PR DESCRIPTION
**Pull request by jacobwhite**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
SearchEngines.swift | 266 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by garvankeeley** [comment]
Ping to our team to take action if someone is familiar with this issue.
I see zero issue with landing this, it is just an added `if let =` which looks to me.

File: Client/Frontend/Browser/SearchEngines.swift [hunk 1]
```diff
@@ -263,9 +263,10 @@ class SearchEngines {
             assert(FileManager.default.fileExists(atPath: fullPath), "\(fullPath) exists")
 
             let engine = parser.parse(fullPath, engineID: engineName)
-            assert(engine != nil, "Engine at \(fullPath) successfully parsed")
-
-            engines.append(engine!)
+            if let engine = parser.parse(fullPath, engineID: engineName) {
+                print (")Engine at \(fullPath) successfully parsed")
```
Comment (garvankeeley) [inline Client/Frontend/Browser/SearchEngines.swift]: There is a stray bracket at the start of the message. I think we only want to print the error case, not the success.

**Comment by farhanpatel** [comment]
There are changes to the way we handle searchEngines that will most likely not require these changes anymore.

**Pull request closed without merging**